command/certificate/needsRenewal.go

```diff
@@ -0,0 +1,163 @@
+package certificate
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/errs"
+	"github.com/urfave/cli"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func needsRenewalCommand() cli.Command {
+	return cli.Command{
+		Name:      "needs-renewal",
+		Action:    cli.ActionFunc(needsRenewalAction),
+		Usage:     `Check if a certificate needs to be renewed`,
+		UsageText: `**step certificate needs-renewal** <crt_file or host_name> [**--expires-in <duration>]`,
+		Description: `**step certificate needs-renewal** Checks certificate expiration from file or from a host if 
+		the certificate is over 66% of its lifetime. If the certificate needs renewal this command will return '0'.
+		If the certificate is not far enough into its life, it will return '1'. If validation fails, or if an error occurs, 
+		this command will produce a non-zero return value.
+## POSITIONAL ARGUMENTS
+
+<crt_file>
+: The path to a certificate to validate.
+
+<host_name>
+: Address of remote host 
+
+## EXIT CODES
+
+This command returns 0 if needing renewal, or returns 1 if the certificate does not need renewal. It will return 255 if any errors occurred
+
+## EXAMPLES
+Check certificate for renewal using custom directory 
+'''
+$ step certificate needs-renewal ./certificate.crt 
+'''
+Check certificate for renewal using a host
+$ step certificate needs-renewal https://smallstep.com
+'''
+Check if certificate will expire within a given time
+$ step certificate needs-renewal ./certificate.crt --expires-in 1h15m
+'''
+Check if certificate from host will expire within a given time
+$ step certificate needs-renewal https://smallstep.com --expires-in 1h15m
+'''
+Check if certificate has passed a percentage of its lifetime 
+$ step certificate needs-renewal ./certificate.crt --expires-in 75%
+'''
+Check if certificate from a host has passed a percentage of its lifetime
+$ step certificate needs-renewal https://smallstep.com --expires-in 75%
+`,
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name: "expires-in",
+				Usage: `Check if the certificate expires in given time duration
+				using <percent|duration>. For <percent>, must be followed by "%".
+				With <duration>, it is a sequence of decimal numbers, each with optional
+				fraction and a unit suffix, such as "300ms", "-1.5h" or "2h45m". Valid
+				time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".`,
+			},
+		},
+	}
+}
+
+func needsRenewalAction(ctx *cli.Context) error {
+	if err := errs.NumberOfArguments(ctx, 1); err != nil {
+		return err
+	}
+	var (
+		crtFile    = ctx.Args().Get(0)
+		expiresIn  = ctx.String("expires-in")
+		roots      = ctx.String("roots")
+		serverName = ctx.String("servername")
+		cert       *x509.Certificate
+	)
+
+	if addr, isURL, err := trimURL(crtFile); err != nil {
+		return err
+	} else if isURL {
+		peerCertificates, err := getPeerCertificates(addr, serverName, roots, false)
```

Comment:
Since we're changing the operation of this command to now apply to all certificates in the bundle, you can probably just use the full list of peer certificates rather than grabbing the first index.